### PR TITLE
Fix CRD reconsiling for cluster-api types

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by alvaroaleman** [comment]
Two sidenotes to add here:

a) I discovered by accident that setting flags like `- --flag-name=flag-val` instead of `- --flag-name\n- flag-val` works just as well - Do we want to change this globally as it IMHO improves readability of the flag setting by a lot?

b) There is still #1351 open which was scheduled for 2.8 but that is too late now - Do we want to put that in the 2.9 cycle and also remove the the `Cluser.Spec.MasterVersion` field (which implies using a different cluster type for the migration tool as it otherwise cant migrate that field anymore)?

File: api/pkg/controller/cluster/userclusterresources.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ import (
 	"github.com/kubermatic/kubermatic/api/pkg/resources/machinecontroller"
 	"github.com/kubermatic/kubermatic/api/pkg/resources/vpnsidecar"
 
+	"github.com/Masterminds/semver"
```
Comment (mrIncompetent) [inline api/pkg/controller/cluster/userclusterresources.go]: Should be grouped with imports on https://github.com/kubermatic/kubermatic/pull/2095/files#diff-2423f6032f1bd40f9a910fad46eda1e5R7

File: api/pkg/controller/cluster/userclusterresources.go [hunk 2]
```diff
@@ -382,9 +384,14 @@ func (cc *Controller) userClusterEnsureCustomResourceDefinitions(c *kubermaticv1
...
 
+	version, err := semver.NewVersion(c.Spec.Version)
+	if err != nil {
+		return fmt.Errorf("failed to extract version from cluster %s: %v", c.Name, err)
+	}
+
 	for _, create := range GetCRDCreators() {
 		var existing *apiextensionsv1beta1.CustomResourceDefinition
-		crd, err := create(nil)
+		crd, err := create(version.Minor(), nil)
```
Comment (mrIncompetent) [inline api/pkg/controller/cluster/userclusterresources.go]: Why not passing in the whole `Version` object?
Just passing the minor seems a bit unintuitive

**Comment by mrIncompetent** [comment]
> a) I discovered by accident that setting flags like - --flag-name=flag-val instead of - --flag-name\n- flag-val works just as well - Do we want to change this globally as it IMHO improves readability of the flag setting by a lot?

We used the latter aproach to avoid having to quote specific flag values. I encountered some issues with the `--flag-name=flag-val` approach - though i can't remember what those issues where...
In terms of readability you're just talking about the readability of the deployed manifest? Or about the go-code as well?

> b) There is still #1351 open which was scheduled for 2.8 but that is too late now - Do we want to put that in the 2.9 cycle and also remove the the Cluser.Spec.MasterVersion field (which implies using a different cluster type for the migration tool as it otherwise cant migrate that field anymore)?

With v2.9 we should be safe with just removing `Cluser.Spec.MasterVersion` & the migration step.
If the `semver.Version` supports json unmarshalling, we could just make `cluster.spec.version` a `*semver.Version` ?

**Comment by alvaroaleman** [comment]
> We used the latter aproach to avoid having to quote specific flag values. I encountered some issues with the --flag-name=flag-val approach - though i can't remember what those issues where...
In terms of readability you're just talking about the readability of the deployed manifest? Or about the go-code as well?

I was just talking about the generated manifests.


WRT the semver discussion: I moved that over to #1351 

Pull was updated, PTAL

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**